cmake/onnxruntime_common.cmake

```diff
@@ -44,6 +44,22 @@ else()
     endif()
 endif()
 
+if(CMAKE_GENERATOR_PLATFORM)
+    # Multi-generator build
+    set(onnxruntime_target_platform ${CMAKE_GENERATOR_PLATFORM})
+else()
+    set(onnxruntime_target_platform ${CMAKE_SYSTEM_PROCESSOR})
+endif()
+if(onnxruntime_target_platform STREQUAL "ARM64")
+    set(onnxruntime_target_platform "ARM64")
+elseif(onnxruntime_target_platform STREQUAL "ARM" OR CMAKE_GENERATOR MATCHES "ARM")
+    set(onnxruntime_target_platform "ARM")
+elseif(onnxruntime_target_platform STREQUAL "x64" OR onnxruntime_target_platform STREQUAL "x86_64" OR onnxruntime_target_platform STREQUAL "AMD64" OR CMAKE_GENERATOR MATCHES "Win64")
+    set(onnxruntime_target_platform "x64")
+elseif(onnxruntime_target_platform STREQUAL "x86" OR onnxruntime_target_platform STREQUAL "i386" OR onnxruntime_target_platform STREQUAL "i686")
+    set(onnxruntime_target_platform "x86")
+endif()
+
```

Comment:
A bit ugly, but we must take into consideration multiple combinations of CMAKE_GENERATOR_PLATFORM (platform in multi-platform generators), CMAKE_SYSTEM_PROCESSOR (target CPU in single platform generators) and CMAKE_GENERATOR

Reply:
Shouldn't this be checking `CMAKE_GENERATOR_PLATFORM MATCHES "ARM"` (line 53, 57), not CMAKE_GENERATOR? CMAKE_GENERATOR will be something like "Visual Studio 15 2017" or "Ninja", not an architecture.

Reply:
For older versions of the VS CMake generator, the platform is part of the generator string. Run `cmake --help`:

```
Generators

The following generators are available on this platform (* marks default):
* Visual Studio 16 2019        = Generates Visual Studio 2019 project files.
                                 Use -A option to specify architecture.
  Visual Studio 15 2017 [arch] = Generates Visual Studio 2017 project files.
                                 Optional [arch] can be "Win64" or "ARM".
  Visual Studio 14 2015 [arch] = Generates Visual Studio 2015 project files.
                                 Optional [arch] can be "Win64" or "ARM".
  Visual Studio 12 2013 [arch] = Generates Visual Studio 2013 project files.
                                 Optional [arch] can be "Win64" or "ARM".
  Visual Studio 11 2012 [arch] = Generates Visual Studio 2012 project files.
                                 Optional [arch] can be "Win64" or "ARM".
```